### PR TITLE
feat: adds generic revalidate paths endpoint

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🤷 Why

Adds a generic `/api/revalidate/paths` endpoint so we can choose particular paths to revalidate. It accepts an array of paths that must be full from the root of the developer domain i.e. `/terraform/docs/some-doc-path`. 

I think this could come in handy in a pinch to run manually if we're having issues with the batch revalidation or in debugging stale content and could also be used in workflows later on. 

## 🧪 Testing

You can test this by sending a post request to the deploy preview url via postman and passing the revalidate token (find in vercel) as a bearer token with the request. 

Visit the logs for the deploy preview and you should see the paths logged as revalidated.

### CODE REVIEW

**Comment by kendallstrautman** [comment]
@BRKalow I have a feeling this will come in handy if we release the 'nuclear' one and have any issues with stale content. So I wanted to pre-emtively get this in.

File: src/pages/api/revalidate-paths.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			const formattedPath = path.replace(/\/$/, '')
+			console.log('[revalidate]', formattedPath)
+			revalidatePromises.push(response.revalidate(formattedPath))
+		})
+
+		await Promise.allSettled(revalidatePromises)
+
+		response
+			.status(StatusCodes.OK)
+			.send(`[revalidation success]: for ${paths.length} tutorial paths`)
```
Comment (kendallstrautman) [inline src/pages/api/revalidate-paths.ts:58]:
```suggestion
			.send(`[revalidation success]: for ${paths.length} paths`)
```

File: src/pages/api/revalidate-paths.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+import { validateToken } from 'lib/api-validate-token'
+
+const PATHS_LIMIT = 100
```
Comment (BRKalow) [inline src/pages/api/revalidate-paths.ts:10]: This is an arbitrary value for now, right?

**Review by BRKalow — APPROVED** [review]
LGTM! I'm wondering if it'd be good to have an `/api/revalidate/` folder now, but no concerns with the implementation here

**Review by zchsh — APPROVED** [review]
🙌  Looks great to me! Tested via Postman as suggested, got a successful `[revalidation success]: for 1 paths` response.

File: src/pages/api/revalidate-paths.ts (hunk 2 above)
Comment (kendallstrautman) [reply src/pages/api/revalidate-paths.ts:10]: Yes! Definitely open for input here, felt like an okay limit to start.

File: src/pages/api/revalidate-paths.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/**
```
[the remark this answers is not in the document]
Comment (kendallstrautman) [reply src/pages/api/revalidate-paths.ts:1]: 💡 I like that idea. I could start with this one and then make a follow up ticket to move the docs one thats in use over?

[the remark this answers is not in the document]
Comment (BRKalow) [reply src/pages/api/revalidate-paths.ts:1]: @kendallstrautman sgtm!

File: src/pages/api/revalidate-paths.ts (hunk 2 above)
Comment (BRKalow) [reply src/pages/api/revalidate-paths.ts:10]: 128? 🤓 (kidding)

Comment (kendallstrautman) [reply src/pages/api/revalidate-paths.ts:10]: NERD

Comment (kendallstrautman) [reply src/pages/api/revalidate-paths.ts]: it is done 🧙🏻

File: src/pages/api/revalidate.ts [hunk 4]
```diff
@@ -9,6 +9,8 @@ import { validateToken } from 'lib/api-validate-token'
 import { cachedGetProductData } from 'lib/get-product-data'
 
 /**
+ * @TODO move this into the /api/revalidate dir and update the filename to something like 'docs'
```
Comment (kendallstrautman) [inline src/pages/api/revalidate.ts:12]:
```suggestion
 * @TODO move this into the /api/revalidate dir and update the filename to something like 'docs'
 * https://app.asana.com/0/1202097197789424/1204094036463681
```

**Pull request merged**